### DISCUSSION TITLE
How to extract a field after applying a transformation

### DISCUSSION THREAD

**Discussion by EnxEng** [GeoStat-Framework/GSTools#168] [Q&A] [answered]
Hi to all, I am new here, great package!
I am having some problems with using this package. Here is my problem:
I am not able to obtain the numerical value of a field, after applying a transformation

```python
import gstools as gs
x = y = range(100)
model = gs.Gaussian(dim=2, var=1, len_scale=10)
srf = gs.SRF(model, seed=20170519)
field = srf.structured([x, y])
srf.plot() #ok is Gaussian
print('Gaussian:')
print(field)
gs.transform.normal_to_lognormal(srf)
srf.plot() #ok is log normal
print('Gaussian (should be lognormal?):')
print(field)
field = srf.structured([x, y])
print('Gaussian (should be lognormal?):')
print(field)
```
All three field prints get the same result.
I would expect to have the field values. srf is overwritten when applying transform... where do I access the field property that has changed?

**Comment by LSchueler** [comment, marked as answer]
Hi EnxEng!

Thanks for the feedback, very much appreciated!

Try this code:

``` python
import gstools as gs

x = y = range(100)
model = gs.Gaussian(dim=2, var=1, len_scale=10)
srf = gs.SRF(model, seed=20170519)
field = srf.structured([x, y])
srf.plot() #ok is Gaussian
print('Gaussian:')
print(field)
gs.transform.normal_to_lognormal(srf)
field_log_norm = srf.field # the transformation is saved internally
srf.plot() #ok is log normal
print('Log-Normal:')
print(field_log_norm)
field_gau_again = srf.structured([x, y])
print('Gaussian again, the srf.structured method overwrites the internally saved transformed field:')
print(field_gau_again)
```


Is that what you are looking for?

@MuellerSeb Is there a good reason why the transformations do not return the transformed field?

**Reply by MuellerSeb** [reply]
@LSchueler good point! Transformations should indeed return the transformed field. Will come up with a PR.

@EnxEng another possibilty, if you only want a log-normal field, is to use the new normalizer classes to apply a (power)tranform inplace:
```python
import gstools as gs
x = y = range(100)
model = gs.Gaussian(dim=2, var=1, len_scale=10)
srf = gs.SRF(model, seed=20170519, normalizer=gs.normalizer.LogNormal)
field = srf.structured([x, y])
srf.plot()  # already transformed
```
Thanks for using GSTools!

Sebastian

**Reply by EnxEng** [reply]
Wow, great! Yes, I was looking for srf.field 
Thank you @MuellerSeb
